From f503ee47ea9eefe828ad5d800ceda0e43192cc46 Mon Sep 17 00:00:00 2001
From: WeiguangTWK <weiguangtwk@outlook.com>
Date: Fri, 3 Jul 2026 16:58:43 +0800
Subject: [PATCH] Neko LineageOS: Face unlock

This feature is developed by AOSPA & CrDroid, check [origional topic](https://review.lineageos.org/q/topic:%2223fu%22)
, for detailed integration check [README](https://github.com/WeiguangTWK/patches_for_build_marble_AOSP/blob/lineage-23.2/README.md)

weiguangtwk:
 - adapt to LineageOS 23.2

Co-authored-by: Francescodario Cuzzocrea <bosconovic@gmail.com>
Co-authored-by: Chris Crump <chriscrawford893@gmail.com>
Change-Id: I8a2477dff8948a0d9a80434cd28761c15aaaa6ce
---
 config/common.mk | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/config/common.mk b/config/common.mk
index 96db7a6..dbb16f2 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -248,6 +248,18 @@ PRODUCT_ARTIFACT_PATH_REQUIREMENT_ALLOWED_LIST += \
 endif
 endif
 
+# Face Unlock
+ifeq ($(TARGET_SUPPORTS_64_BIT_APPS),true)
+PRODUCT_PACKAGES += \
+    FaceUnlock
+
+PRODUCT_SYSTEM_EXT_PROPERTIES += \
+    ro.face.sense_service=true
+
+PRODUCT_COPY_FILES += \
+    frameworks/native/data/etc/android.hardware.biometrics.face.xml:$(TARGET_COPY_OUT_SYSTEM_EXT)/etc/permissions/android.hardware.biometrics.face.xml
+endif
+
 # SystemUI
 PRODUCT_DEXPREOPT_SPEED_APPS += \
     CarSystemUI \
-- 
2.43.0

